readme.md

```diff
@@ -295,13 +295,64 @@ Returns `true` if `value` is an even integer.
 
 Returns `true` if `value` is an odd integer.
 
-##### .empty(value)
+##### .emptyArray(value)
 
-Returns `true` if `value` is falsy or an empty string, array, object, map, or set.
+Returns `true` if the value is an array and the `length` is 0.
 
-##### .emptyOrWhitespace(value)
+##### .nonEmptyArray(value)
 
-Returns `true` if `is.empty(value)` or a string that is all whitespace.
+Returns `true` if the value is an array and the `length` is *not* 0.
+
+##### .emptyString(value)
+
+Returns `true` if the value is an empty string. `false` otherwise.
+
+##### .nonEmptyString(value)
+
+Returns `true` if the value is string, and the length !== 0. `false` when the string is empty.
+
+##### .emptyObject(value)
+
+Returns `true` if the value is an object and `Object.keys(value)` length is 0. 
+
+Please note that `Object.keys` returns only enumerable properties. Hence something like this can happen - 
+
+```js
+const object1 = {};
+
+Object.defineProperty(a, 'property1', {
+	value: 42,
+	writable: true,
+	enumerable: false,
+	configurable: true
+});
+
+is.emptyObject(object1);
+// => true
+```
+##### .nonEmptyObject(value) 
+
+Returns `true` if the value is an object and length of `Object.keys(value)`  is !== 0. `false` if it is 0.
+
+##### .emptySet(value)
+
+Returns `true` if the value is an empty Set (`size` is 0). `false` if it is not zero.
+
+##### .nonEmptySet(Value)
+
+Returns `true` of the value is a Set and it's `size` is not 0. `false` if it is `size` is 0.
+
+##### .emptyMap(value)
+
+Returns `true` if the value is an empty Map (`size` is 0). `false` if it is not zero.
+
+##### .nonEmptyMap(value)
+
+Returns `true` if the value is a Map and it's `size` is not 0. `false` if is 0.
+
+##### .emptyStringOrWhitespace(value)
```

Comment:
Move this up, after the empty and non empty string methods

Reply:
Ok.